inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp

```diff
@@ -685,12 +685,93 @@ layout layout_optimizer::get_expected_layout(layout const& current_layout,
     auto expected_data_type = current_layout.data_type;
     auto expected_format = current_layout.format;
     auto input_layout = node.get_dependency(0).get_output_layout();
+    auto output_layout = node.calc_output_layout();
 
     const float cond_denom = _total_conv > 0 ? 1.0f / static_cast<float>(_total_conv) : 1.0f;
 
-    auto output_layout = node.calc_output_layout();
+    bool is_dw = input_layout.size.feature[0] == static_cast<int>(prim->groups);
+    int ofm_per_group = output_layout.size.feature[0] / prim->groups;
+    int ifm_per_group = input_layout.size.feature[0] / prim->groups;
+    int compute_block = 32;
+    bool valid_grouped = !is_dw && prim->groups > 1 && (ofm_per_group % compute_block == 0 &&  ifm_per_group % compute_block == 0);
+    bool valid_int8_dw = is_dw && output_layout.size.batch[0] % 16 == 0;
+    bool non_grouped = prim->groups == 1;
+    bool is_2d = input_layout.format.spatial_num() == 2;
+    bool valid_post_ops = /* get_post_ops_count(node) <= 32 */ false;
+    bool use_onednn_impls = _optimization_attributes.use_onednn_impls;
+
+    if (use_onednn_impls) {
+        for (auto& fo : node.get_fused_primitives()) {
+            if (fo.node->is_type<eltwise>()) {
+                auto in_layout = node.get_dependency(fo.dep_start_idx).get_output_layout();
+                auto out_layout = node.get_output_layout();
+                auto in_dt = in_layout.data_type;
+                auto out_dt = out_layout.data_type;
+                if ((out_layout.count() == in_layout.count()) &&
+                    (data_type_traits::is_floating_point(in_dt) || data_type_traits::is_floating_point(out_dt)) && in_dt != out_dt)
+                    valid_post_ops = false;
+            }
+        }
+    }
+
+    /* ***************************************************************************************** */
+    /* ***************************** OneDNN layout selection part ****************************** */
+    /* ***************************************************************************************** */
 
-    if ((input_layout.data_type == data_types::u8 || input_layout.data_type == data_types::i8)) {
+    if (use_onednn_impls && (input_layout.data_type == data_types::u8 || input_layout.data_type == data_types::i8)) {
```

Comment:
I think this would look better if we group the code more explicitly, like...

```
if (use_onednn_impls) {
    /* ***************************************************************************************** */
    /* ***************************** OneDNN layout selection part ****************************** */
    /* ***************************************************************************************** */
    if (input_layout.data_type == data_types::u8 || input_layout.data_type == data_types::i8) {
      ...
    }
} else {
    /* ***************************************************************************************** */
    /* *************************** Native impls layout selection part ************************** */
    /* ***************************************************************************************** */
}
```

Reply:
I thought about such approach. But I would like to keep default behavior in case if no one "OneDNN friendly" conditions are met. That`s why I decided to use single if-else statement for both scenarios

Reply:
I dared to slightly modify this part as Mingyu suggested initially
Could you please take a look, guys? @isanghao , @vladimir-paramuzov, @lznamens